### DISCUSSION TITLE
external access to image data

### DISCUSSION THREAD

**Discussion by Yamioni2007** [nextcloud/recognize#845] [Q&A] [answered]
Hi 

How can I access image data in an **external** way over a Web API/ wget via webdav/ or similar? (Of course in combination with a admin/user password)

I'm especially looking for a solution without writing my own API in PHP.

A little bit of context:

I would like to build an extension to speed up recognize a little bit. The idea is to take a few objects from the queue, process them on a different (faster) machine and put the results back.

I already found the table oc_recognize_queue_faces (with file_id, storage_id, root_id) and know how to access it, the problem is to get access to the image data itself connected to this table.

Thanks in advance

**Comment by marcelklehr** [comment, marked as answer]
> I would like to build an extension to speed up recognize a little bit.

Cool idea! I'll try to help with advice whenever you need it :)

You could use the preview endpoint: /core/preview?fileId=XXXXX&x=-1&y=384&a=1
